packages/flutter/lib/src/widgets/async.dart

```diff
@@ -520,11 +628,21 @@ class FutureBuilder<T> extends StatefulWidget {
   /// Creates a widget that builds itself based on the latest snapshot of
   /// interaction with a [Future].
   ///
-  /// The [builder] must not be null.
+  // ignore: deprecated_member_use_from_same_package
+  /// The [initialData] argument specifies the data that will be used to create
+  /// the snapshots provided to [builder] until a non-null [future] has
+  /// completed.
+  ///
+  /// The [builder] argument must not be null.
   const FutureBuilder({
     Key key,
     this.future,
-    this.initialData,
+    @Deprecated(
+      "Don't provide initialData to FutureBuilder. Instead, check for "
```

Comment:
we generally would phrase this more softly as "Instead of providing initialData to a FutureBuilder, consider checking for..."